dist/timesync.js

```diff
@@ -862,7 +862,11 @@ Promise.disableSynchronous = function() {
 },{"./core.js":4}],11:[function(require,module,exports){
 'use strict';
 
-module.exports = typeof window === 'undefined' || typeof window.Promise === 'undefined' ? require('promise') : window.Promise;
+var selfPromise = typeof self !== 'undefined' && typeof self.Promise !== 'undefined' && self.Promise;
```

Comment:
quick question: why do we need this? All major browsers have promises, and node has it well. couldn't we simply use the global Promise (available both in the window and the node global object)?

Reply:
no idea, wanted to keep backward compatibility.

Reply:
I don't think that should be a concern. Like I said, all major browsers support promises now.